src/common/src/types/mod.rs

```diff
@@ -961,6 +1016,9 @@ impl ScalarRefImpl<'_> {
             Self::Jsonb(v) => v.memcmp_serialize(ser)?,
             Self::Struct(v) => v.memcmp_serialize(ser)?,
             Self::List(v) => v.memcmp_serialize(ser)?,
+            // Map should not be used as key.
+            // This should be banned in frontend and this branch should actually be unreachable.
```

Comment:
Is this (banning it from being a key) still planned? Shall we add error UI tests for it (in future PRs)?

Reply:
Yes. But I'm thinking there's no need to ban it here now, since it has a defined semantics.